### PR TITLE
Add show route

### PR DESCRIPTION
**Pull request by skeltont**
This branch implements the ability to view an individual route's detail, edit it if you're the owner, and add comments! 

It also brings a little housekeeping with the interactions between routes/assigned_routes controllers along with some naming changes to the latter.

### CODE REVIEW

File: app/assets/javascripts/comment.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// Place all the behaviors and hooks related to the matching controller here.
+// All this logic will automatically be available in application.js.
+// You can use CoffeeScript in this file: http://coffeescript.org/
+$(document).ready(function() {
```
Comment (MillerJL) [inline app/assets/javascripts/comment.js]: Might need to pass event into function. Might be completely wrong. Ran into an issue with firefox in the past where this ended up being the cause.

``` javascript
+  $('#comment_comment').keyup(function(event) {
```

http://stackoverflow.com/questions/20522887/referenceerror-event-is-not-defined-error-in-firefox

Comment (jonahgeorge) [inline app/assets/javascripts/comment.js]: This also fails when using Turbolinks. 
http://stackoverflow.com/questions/18770517/rails-4-how-to-use-document-ready-with-turbo-links

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+class CommentsController < ApplicationController
+
+  def index
+  end
+
+  def create
+    @comment = Comment.new
+    ## @HACK: this feels like a hack, so I will come back to this.
```
Comment (skeltont) [inline app/controllers/comments_controller.rb]: this is the change:

``` ruby
def create
    @comment = Comment.new(comment_params)
    @comment.user_id = current_user.id
    @comment.route_id = params[:route_id]
    if @comment.save
      redirect_to request.referer
    end
  end
```

**Comment by jonahgeorge** [comment]
Might be good to put a presence validation on the `comment` field of `comment.rb` in case our javascript errors out and the jQuery hook doesn't fire
http://apidock.com/rails/ActiveRecord/Validations/ClassMethods/validates_presence_of

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  end
+
+  def create
+    @comment = Comment.new(comment_params)
+    @comment.user_id = current_user.id
+    @comment.route_id = params[:route_id]
+    if @comment.save
+      redirect_to request.referer
+    else
+      flash[:danger] = "Couldn't save your comment"
```
Comment (skeltont) [inline app/controllers/comments_controller.rb:13]: took out the word 'shitty'
sorry @jonahgeorge

**Pull request merged**